mp2-common/src/eth.rs

```diff
@@ -111,6 +116,175 @@ pub struct ProofQuery {
     pub(crate) slot: StorageSlot,
 }
 
+/// Struct used for storing relevant data to query blocks as they come in.
+#[derive(Debug, Clone)]
+pub struct ReceiptQuery {
+    /// The contract that emits the event we care about
+    pub contract: Address,
+    /// The event we wish to monitor for,
+    pub event: Event,
+}
+
+/// Struct used to store all the information needed for proving a leaf in the Receipt Trie is one we care about.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ReceiptProofInfo {
+    /// The MPT proof that this Receipt is in the tree
+    pub mpt_proof: Vec<Vec<u8>>,
+    /// The index of this transaction in the block
+    pub tx_index: u64,
+    /// The size of the index in bytes
+    pub index_size: usize,
+    /// The offset in the leaf (in RLP form) to status
+    pub status_offset: usize,
+    /// The offset in the leaf (in RLP form) to the start of logs
+    pub logs_offset: usize,
+    /// Data about the type of log we are proving the existence of
+    pub event_log_info: EventLogInfo,
+    /// The offsets for the relevant logs
+    pub relevant_logs_offset: Vec<usize>,
+}
+
+/// Contains all the information for an [`Event`] in rlp form
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct EventLogInfo {
+    /// Size in bytes of the whole log rlp encoded
+    pub size: usize,
+    /// Packed contract address to check
+    pub address: Address,
+    /// Byte offset for the address from the beginning of a Log
+    pub add_rel_offset: usize,
+    /// Packed event signature,
+    pub event_signature: [u8; 32],
+    /// Byte offset from the start of the log to event signature
+    pub sig_rel_offset: usize,
+    /// The topics for this Log
+    pub topics: [LogDataInfo; 3],
+    /// The extra data stored by this Log
+    pub data: [LogDataInfo; 2],
```

Comment:
Can we put this length  in a const generic ? Maybe one is enough or 2 is not enough.